model.py

```diff
@@ -1211,16 +1218,114 @@ class Hyperlink(Base):
     __tablename__ = 'hyperlinks'
 
     id = Column(Integer, primary_key=True)
-    library_id = Column(Integer, ForeignKey('libraries.id'), index=True)
     rel = Column(Unicode, index=True, nullable=False)
-    href = Column(Unicode, nullable=False)
+    library_id = Column(Integer, ForeignKey('libraries.id'), index=True)
+    resource_id = Column(Integer, ForeignKey('resources.id'), index=True)
 
     # A Library can have multiple links with the same rel, but we only
     # need to keep track of one.
     __table_args__ = (
         UniqueConstraint('library_id', 'rel'),
     )
 
+    @hybrid_property
+    def href(self):
+        if not self.resource:
+            return None
+        return self.resource.href
+
+    @href.setter
+    def href(self, url):
+        _db = Session.object_session(self)
+        resource, is_new = get_one_or_create(_db, Resource, href=url)
+        self.resource = resource
```

Comment:
This hybrid property let me keep most of the old code, which assumes you can read and write Hyperlink.href.